emily/handler/src/database/entries/deposit.rs

```diff
@@ -155,6 +156,46 @@ impl DepositEntry {
             self.key(),
         )))
     }
+
+    /// Reorgs around a given chainstate.
+    /// TODO(TBD): Remove duplicate code around deposits and withdrawals if possible.
+    pub fn reorganize_around(&mut self, chainstate: &Chainstate) -> Result<(), Error> {
+        // Update the history to have the histories wiped after the reorg.
+        self.history.retain(|event| {
+            // The event is younger than the reorg...
+            (chainstate.stacks_block_height > event.stacks_block_height)
+                // Or the event is as old as the reorg but has a different block hash...
+                || ((chainstate.stacks_block_height == event.stacks_block_height)
+                    && (chainstate.stacks_block_hash == event.stacks_block_hash))
+        });
+        // If the history is empty add a reassessing event.
+        if self.history.is_empty() {
+            self.history = vec![DepositEvent {
+                status: StatusEntry::Reprocessing,
```

Comment:
nit: shouldn't this comment say reprocessing event?

Reply:
Done!